boot/bootutil/src/loader.c

```diff
@@ -245,14 +215,18 @@ boot_initialize_area(struct boot_loader_state *state, int flash_area)
         return BOOT_EFLASH;
     }
 
+#ifdef MCUBOOT_USE_FLASH_AREA_GET_SECTORS
     rc = flash_area_get_sectors(flash_area, &num_sectors, out_sectors);
+#else
+    _Static_assert(sizeof(int) <= sizeof(uint32_t), "Fix needed");
```

Comment:
This was officially introduced only in C11, but some earlier compiler versions already support it. Since it already appears in the code and nobody has complained, it is fine. I have not found any statement in the repo that which C version is supported officially.

Reply:
It kind of depends on the environment used to build MCUboot, which is fairly diverse. Given the instances of C11 features, perhaps that is adequate. We also have other habits, such as declaring all variables at the top of functions that hasn't been necessary since C89, so who knows what we are targeting.